sumologic/resource_sumologic_local_file_source_test.go

```diff
@@ -0,0 +1,259 @@
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckLocalFileSourceDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "sumologic_localFile_source" {
+			continue
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("LocalFile Source destruction check: LocalFile Source ID is not set")
+		}
+
+		id, err := strconv.Atoi(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+
+		collectorID, err := strconv.Atoi(rs.Primary.Attributes["collector_id"])
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+
+		s, err := client.GetLocalFileSource(collectorID, id)
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		if s != nil {
+			return fmt.Errorf("LocalFile Source still exists")
+		}
+	}
+	return nil
+}
+
+func testAccCheckLocalFileSourceExists(n string, localFileSource *LocalFileSource) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("LocalFile Source ID is not set")
+		}
+
+		id, err := strconv.Atoi(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("LocalFile Source id should be int; got %s", rs.Primary.ID)
+		}
+
+		collectorID, err := strconv.Atoi(rs.Primary.Attributes["collector_id"])
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+
+		c := testAccProvider.Meta().(*Client)
+		localFileSourceResp, err := c.GetLocalFileSource(collectorID, id)
+		if err != nil {
+			return err
+		}
+
+		*localFileSource = *localFileSourceResp
+
+		return nil
+	}
+}
+
+func testAccCheckLocalFileSourceValues(localFileSource *LocalFileSource, name, description, category string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if localFileSource.Name != name {
+			return fmt.Errorf("bad name, expected \"%s\", got: %#v", name, localFileSource.Name)
+		}
+		if localFileSource.Description != description {
+			return fmt.Errorf("bad description, expected \"%s\", got: %#v", description, localFileSource.Description)
+		}
+		if localFileSource.Category != category {
+			return fmt.Errorf("bad category, expected \"%s\", got: %#v", category, localFileSource.Category)
+		}
+		return nil
+	}
+}
+
+func testAccSumologicLocalFileSourceConfig(cName, cDescription, cCategory, sName, sDescription, sCategory, tName, tDescription, tCategory, kName, kDescription, kCategory string) string {
+	return fmt.Sprintf(`
+resource "sumologic_collector" "test" {
+	name = "%s"
+	description = "%s"
+	category = "%s"
+}
+	
+resource "sumologic_local_file_source" "localFile" {
+	name = "%s"
+	description = "%s"
+	category = "%s"
+	collector_id = "${sumologic_collector.test.id}"
+}
```

Comment:
I believe we need more fields to be added here (at least the required field `path_expression`) so the source creation will not fail

Reply:
Also, the `sumologic_collector` is for the hosted collector. We would need to have a new resource for `installed_collector` as the `sumologic_local_file_source` is an Installed collector source.

Refer: https://github.com/SumoLogic/terraform-provider-sumologic/pull/182

Reply:
Interesting, I may have need to leverage that installed_collector construct when it's deployed.